packages/libs/error-stack/src/fmt/mod.rs

```diff
@@ -738,16 +749,27 @@ fn debug_attachments(
     let (next, defer): (Vec<_>, _) = frames
         .into_iter()
         .map(|frame| match frame.kind() {
-            FrameKind::Context(_) => unreachable!(),
-            FrameKind::Attachment(AttachmentKind::Opaque(_)) => {
+            FrameKind::Attachment(AttachmentKind::Opaque(_)) | FrameKind::Context(_) => {
+                #[cfg(all(nightly, feature = "unstable"))]
+                if let Some(debug) = frame.request_value::<DebugDiagnostic>() {
+                    let (emit, snippets) = debug.into_parts();
+
+                    for snippet in snippets {
+                        ctx.as_hook_context::<DebugDiagnostic>()
+                            .attach_snippet(snippet);
+                    }
+
+                    return emit;
+                }
+
                 #[cfg(all(nightly, feature = "unstable"))]
                 if let Some(debug) = frame.request_ref::<DebugDiagnostic>() {
                     for snippet in debug.snippets() {
                         ctx.as_hook_context::<DebugDiagnostic>()
                             .attach_snippet(snippet.clone());
                     }
 
-                    return Some(debug.output().clone());
+                    return debug.emit().iter().cloned().collect::<Vec<_>>();
```

Comment:
```suggestion
                    return debug.emit().iter().to_vec();
```

Reply:
fixed in [1edf76f](https://github.com/hashintel/hash/pull/974/commits/1edf76fc1a37c4f0bda0ee7d44d8c253e6d15a04)